### ISSUE TITLE
Binary is broken if pandas and pyarrow is installed

### ISSUE THREAD

**Issue by JanSellner** [Nuitka/Nuitka#1755]
I am currently using nuitka to create an executable for a simple Python script and so far I am really impressed by nuitka.

I am using pandas in my script and I am facing the issue that as soon as I have [`pyarrow`](https://arrow.apache.org/docs/python/index.html) in my environment, the resulting binary from nuitka does not work anymore. I can run it, but it does nothing. Here is a script to reproduce the issue (save as `hello.py`):
```python
import pandas as pd

def talk(message):
    return "Talk " + message


def main():
    print(talk("Hello World"))


if __name__ == "__main__":
    print(pd.__name__)
    main()
```

Steps to reproduce:
```
conda create --yes --name nuitka python=3.10
conda activate nuitka
pip install nuitka zstandard ordered-set pandas
pip install pyarrow
python -m nuitka --enable-plugin=numpy --nofollow-import-to=unittest --nofollow-import-to=matplotlib --standalone --onefile hello.py
```

There is no output when running `.\hello.exe`. However, when the `pyarrow` step above is skipped, then it works as expected:
```
pandas
Talk Hello World
```

`python -m nuitka --version`:
```
1.0.5
Commercial: None
Python: 3.10.4 | packaged by conda-forge | (main, Mar 30 2022, 08:38:02) [MSC v.1916 64 bit (AMD64)]
Flavor: Anaconda Python
Executable: C:\Users\Jan\miniconda3\envs\nuitka\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```

I am currently working around this issue by saving my tables as csv instead of feather and removing pyarrow first from my environment but would of course be cool if this worked out of the box :-)

**Comment by kayhayen** [comment]
I thought pyarrow works,can you try with CPython, just to check if it is related to using Anaconda?

**Comment by JanSellner** [comment]
You are right! I just tested in a Windows Sandbox with the default Python interpreter and it worked indeed. I am a bit surprised, I thought conda uses the same Python interpreter just in an environment?

```
python.exe -m nuitka --version
1.0.5
Commercial: None
Python: 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Flavor: Unknown
Executable: C:\Users\WDAGUtilityAccount\AppData\Local\Programs\Python\Python310\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
```

**Comment by kayhayen** [comment]
Conda, much like Debian modifies software a lot. One goal is to make environments share stuff. So Anaconda will put libraries into unexpected places, hard code their loading, expect or react to environment variable names, and push these changes sometimes even upstream, so e.g. compiled programs then misbehave when an environment happens to be active. It's pretty bad actually. I will look into this, but maybe I better starting documenting how to produce the Yaml to repair the damage they did.

**Comment by kayhayen** [comment]
I think for DLLs with Anaconda, there are still issues. I would need you to retry with current develop version and Anaconda, and tell me if it is better. I think base environment and conda environment were used in wrong order, and that got fixed there, but now that I look at, only for macOS so far. I need to visit Anaconda on Linux with this mind I guess.
